lib/Composer.php

```diff
@@ -60,6 +71,17 @@ public static function postUpdate(Event $event)
         $rootPath = self::getRootPath($event);
         self::parametersYmlCheck($rootPath);
         self::zendFrameworkOptimization($rootPath);
+        self::executeMigrationsUp($event);
```

Comment:
is it really a good idea to run migrations on composer update? Sometimes there is no database available during this step like when building docker containers with the source-code included. Database would then only be available on runtime of the container.

Reply:
If it fails during the composer update nothing happens (except an error message) you can perform the migrations anytime after the upgrade with the following command: 
`./bin/console pimcore:migrations:migrate -s pimcore_core -n` 
Other projects are doing it the same way, I don't see any problems with that so far. 

Btw. you are free to modify your `composer.json`, if you're having problems with that, just remove the event listener and run the scripts manually.

Reply:
Our aim is to make the upgrading procedure as simple as possible, but still giving the possibility to customize it for your needs ... I think with the current solution we're fulfilling both topics.

Reply:
if it fails during the update, and exit code != 0 happens, which means for docker (or other deployment solutions) that something failed.

Modifying the `composer.json` isn't a real solution, cause it removes other Pimcore tasks from composer as well.

Reply:
Only the `parameters.yml` check and the deprecated ZF optimization, ... but anyway :)  
If it makes you feeling more comfortable, we can move the migrations into a separate `post-update-cmd` part - I can live with that ;-)

Reply:
And running two commands: 

- ```composer update``` and 
- ```bin/console pimcore:migrations:run```   (just an example)

isn't too bad. In other Symfony projects, you first need to copy migrations files from different Bundles to run them, so having Pimcore Migrations is a huge win anyway.

Reply:
@brusch Sounds like a compromise I can live with :)

Reply:
@brusch And regarding the other tasks, you don't know today what will change there. Maybe more tasks will come and changing that then will get complicated ;)

Reply:
@dpfaffenbauer I think splitting them up makes sense anyway, because the migrations should run after the symfony scripts not prior them.

Reply:
alright, happy with that 👍